File: 프로그래머스/unrated/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    storey = storey.toString();
    let result = Infinity;

    (dfs = (pivot, acc, add) => {
        if (pivot < 0) {
            result = Math.min(result, acc + add);
            return result;
        }

        const num = parseInt(storey[pivot]) + add;
        dfs(pivot - 1, acc + num, 0);
        dfs(pivot - 1, acc + 10 - num, 1);
    })(storey.length - 1, 0, 0);
    return result;
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int storey) {
    // 24.12.18
    int answer = 0;
    while(storey)
    {
        int n = storey % 10;
        storey /= 10;
        
        bool isUp = (n == 5 && storey % 10 >= 5) || (n > 5);  
        answer += isUp ? (10 - n) : n; 
        storey += isUp ? 1 : 0;
    }
    return answer;
}